### ISSUE TITLE
Bug: FIREWALL_OUTBOUND_SUBNETS IPv6 and IPv4 family mix

### ISSUE THREAD

**Issue by Lynnesbian** [qdm12/gluetun#1706]
### Is this urgent?

No

### Host OS

Debian Trixie

### CPU arch

x86_64

### VPN service provider

ProtonVPN

### What are you using to run the container

Podman

### What is the version of Gluetun

Running version pr-1543 built on 2023-06-06T09:12:09.468Z (commit f58c459)

### What's the problem 🤔

I'm using [this PR](https://github.com/qdm12/gluetun/pull/1543) rather than the latest official version.

When setting the `FIREWALL_OUTBOUND_SUBNETS` environment variable, the gluetun container fails to start, with an error regarding IPv6 firewalls.

In my case, the variable was set to `172.22.0.0/16,10.0.0.0/24`

### Share your logs

```plain text
2023-06-28T01:00:51+10:00 INFO [routing] default route found: interface tap0, gateway 10.0.2.2, assigned IP 10.0.2.100 and family v4
2023-06-28T01:00:51+10:00 INFO [routing] default route found: interface tap0, gateway fe80::2, assigned IP fd00::607b:aaff:fe77:5874 and family v6
2023-06-28T01:00:51+10:00 INFO [routing] adding route for 0.0.0.0/0
2023-06-28T01:00:51+10:00 INFO [routing] adding route for ::/0
2023-06-28T01:00:51+10:00 INFO [firewall] setting allowed subnets...
2023-06-28T01:00:51+10:00 INFO [routing] routing cleanup...
2023-06-28T01:00:51+10:00 INFO [routing] default route found: interface tap0, gateway 10.0.2.2, assigned IP 10.0.2.100 and family v4
2023-06-28T01:00:51+10:00 INFO [routing] default route found: interface tap0, gateway fe80::2, assigned IP fd00::607b:aaff:fe77:5874 and family v6
2023-06-28T01:00:51+10:00 INFO [routing] deleting route for 0.0.0.0/0
2023-06-28T01:00:51+10:00 INFO [routing] deleting route for ::/0
2023-06-28T01:00:51+10:00 ERROR setting allowed outbound subnets: command failed: "ip6tables-nft --append OUTPUT -o tap0 -s fd00::607b:aaff:fe77:5874 -d 10.0.0.0/24 -j ACCEPT": ip6tables v1.8.9 (nf_tables): host/network `10.0.0.0' not found
Try `ip6tables -h' or 'ip6tables --help' for more information.: exit status 2
```

### Share your configuration

```yml
# container-gluetun.service
# autogenerated by Podman 4.5.1
# Tue Jun 27 23:57:28 AEST 2023

[Unit]
Description=Podman container-gluetun.service
Documentation=man:podman-generate-systemd(1)
Wants=network-online.target
After=network-online.target
RequiresMountsFor=%t/containers

[Service]
Environment=PODMAN_SYSTEMD_UNIT=%n
Restart=on-failure
TimeoutStopSec=70
ExecStartPre=/bin/rm -f %t/%n-pid %t/%n-cid

ExecStart=/usr/bin/podman run \
	--cidfile=%t/%n-cid \
	--conmon-pidfile=%t/%n-pid \
	--cgroups=no-conmon \
	--rm \
	-d \
	--replace \
	--name=gluetun \
	--cap-add NET_ADMIN \
	-e TZ=Australia/Brisbane \
	-e PUID=1000 \
	-e GUID=1000 \
	-e VPN_SERVICE_PROVIDER=protonvpn \
	-e OPENVPN_USER=[removed] \
	-e OPENVPN_PASSWORD=[removed] \
	-e VPN_ENDPOINT_IP=[removed] \
	-e VPN_PORT_FORWARDING=on \
	-e FIREWALL_OUTBOUND_SUBNETS=172.22.0.0/16,10.0.0.0/24 \
	--device /dev/net/tun:/dev/net/tun \
	--privileged \
	--group-add keep-groups \
	-p 6687:6687 \
	-p 6688:6688 \
	-v /home/pool/Documents/podconf/gluetun:/tmp/gluetun:z \
	docker.io/qmcgaw/gluetun:pr-1543

ExecStop=/usr/bin/podman stop \
	--ignore -t 10 \
	--cidfile=%t/%n-cid
ExecStopPost=/usr/bin/podman rm \
	-f \
	--ignore -t 10 \
	--cidfile=%t/%n-cid
PIDFile=%t/%n-pid
Type=forking

[Install]
WantedBy=default.target
```

**Comment by qdm12** [comment]
I rebased that branch on the master branch, can you try pulling that pr image again and see if it's resolved now?

**Comment by Lynnesbian** [comment]
It's fixed now, thank you!

**Comment by Lynnesbian** [comment]
Whoops, my mistake - I had removed the environment variable. Readding it and pulling the latest version still shows the error.

**Comment by qdm12** [comment]
And does this work on the v3.34.3 / v3.35.0 / latest image? I would guess it fails too right?

**Comment by Lynnesbian** [comment]
I just tried the latest image (`docker.io/qmcgaw/gluetun`) - the issue still occurs. Additionally, I had to remove the line `-e VPN_PORT_FORWARDING=on`, although that is expected behaviour.

**Comment by qdm12** [comment]
I'm confused, does the latest image with `VPN_PORT_FORWARDING=off` work for you? Does it only fail when this variable is set to `on`?

Can you also please try `docker.io/qmcgaw/gluetun:v3.34.3` and `docker.io/qmcgaw/gluetun:v3.35.0` to see if it works or not, in order to pinpoint where the problem came from better.

**Comment by kevinkellner** [comment]
I have the same issue.

`VPN_PORT_FORWARDING=off` or `VPN_PORT_FORWARDING=on` does not make a difference. 

The error changes if I use v3.34.3:

`Creating gluetun ... done
Attaching to gluetun
gluetun    | ========================================
gluetun    | ========================================
gluetun    | =============== gluetun ================
gluetun    | ========================================
gluetun    | =========== Made with ❤️ by ============
gluetun    | ======= https://github.com/qdm12 =======
gluetun    | ========================================
gluetun    | ========================================
gluetun    | 
gluetun    | Running version v3.34.3 built on 2023-05-31T14:57:40.709Z (commit 42caa64)
gluetun    | 
gluetun    | 🔧 Need help? https://github.com/qdm12/gluetun/discussions/new
gluetun    | 🐛 Bug? https://github.com/qdm12/gluetun/issues/new
gluetun    | ✨ New feature? https://github.com/qdm12/gluetun/issues/new
gluetun    | ☕ Discussion? https://github.com/qdm12/gluetun/discussions/new
gluetun    | 💻 Email? quentin.mcgaw@gmail.com
gluetun    | 💰 Help me? https://www.paypal.me/qmcgaw https://github.com/sponsors/qdm12
gluetun    | 2023-07-01T09:01:33+02:00 INFO [routing] default route found: interface eth0, gateway 172.18.0.1, assigned IP 172.18.0.26 and family v4
gluetun    | 2023-07-01T09:01:33+02:00 INFO [routing] default route found: interface eth0, gateway fdb3:cca5:e4ba:1000:d0cc:1:0:1, assigned IP 172.18.0.26 and family v6
gluetun    | 2023-07-01T09:01:33+02:00 INFO [routing] local ethernet link found: eth1
gluetun    | 2023-07-01T09:01:33+02:00 INFO [routing] local ethernet link found: eth0
gluetun    | 2023-07-01T09:01:33+02:00 INFO [routing] local ipnet found: 10.5.0.0/16
gluetun    | 2023-07-01T09:01:33+02:00 INFO [routing] local ipnet found: 172.18.0.0/16
gluetun    | 2023-07-01T09:01:33+02:00 INFO [routing] local ipnet found: fdb3:cca5:e4ba:1000:d0cc:1::/112
gluetun    | 2023-07-01T09:01:33+02:00 INFO [routing] local ipnet found: fe80::/64
gluetun    | 2023-07-01T09:01:33+02:00 INFO [firewall] enabling...
gluetun    | 2023-07-01T09:01:33+02:00 INFO [firewall] enabled successfully
gluetun    | 2023-07-01T09:01:34+02:00 INFO [storage] merging by most recent 13056 hardcoded servers and 17657 servers read from /gluetun/servers.json
gluetun    | 2023-07-01T09:01:34+02:00 INFO [storage] Using mullvad servers from file which are 96 days more recent
gluetun    | 2023-07-01T09:01:34+02:00 INFO [storage] Using nordvpn servers from file which are 393 days more recent
gluetun    | 2023-07-01T09:01:34+02:00 INFO Alpine version: 3.17.3
gluetun    | 2023-07-01T09:01:34+02:00 INFO OpenVPN 2.4 version: 2.4.12
gluetun    | 2023-07-01T09:01:34+02:00 INFO OpenVPN 2.5 version: 2.5.8
gluetun    | 2023-07-01T09:01:34+02:00 INFO Unbound version: 1.17.1
gluetun    | 2023-07-01T09:01:34+02:00 INFO IPtables version: v1.8.8
gluetun    | 2023-07-01T09:01:34+02:00 INFO Settings summary:
gluetun    | ├── VPN settings:
gluetun    | |   ├── VPN provider settings:
gluetun    | |   |   ├── Name: mullvad
gluetun    | |   |   └── Server selection settings:
gluetun    | |   |       ├── VPN type: wireguard
gluetun    | |   |       ├── Cities: prague
gluetun    | |   |       └── Wireguard selection settings:
gluetun    | |   └── Wireguard settings:
gluetun    | |       ├── Private key: ML...30=
gluetun    | |       ├── Interface addresses:
gluetun    | |       |   └── 10.64.252.88/32
gluetun    | |       └── Network interface: tun0
gluetun    | ├── DNS settings:
gluetun    | |   ├── DNS server address to use: 127.0.0.1
gluetun    | |   ├── Keep existing nameserver(s): no
gluetun    | |   └── DNS over TLS settings:
gluetun    | |       ├── Enabled: yes
gluetun    | |       ├── Update period: every 24h0m0s
gluetun    | |       ├── Unbound settings:
gluetun    | |       |   ├── Authoritative servers:
gluetun    | |       |   |   └── cloudflare
gluetun    | |       |   ├── Caching: yes
gluetun    | |       |   ├── IPv6: no
gluetun    | |       |   ├── Verbosity level: 1
gluetun    | |       |   ├── Verbosity details level: 0
gluetun    | |       |   ├── Validation log level: 0
gluetun    | |       |   ├── System user: root
gluetun    | |       |   └── Allowed networks:
gluetun    | |       |       ├── 0.0.0.0/0
gluetun    | |       |       └── ::/0
gluetun    | |       └── DNS filtering settings:
gluetun    | |           ├── Block malicious: yes
gluetun    | |           ├── Block ads: no
gluetun    | |           ├── Block surveillance: no
gluetun    | |           └── Blocked IP networks:
gluetun    | |               ├── 127.0.0.1/8
gluetun    | |               ├── 10.0.0.0/8
gluetun    | |               ├── 172.16.0.0/12
gluetun    | |               ├── 192.168.0.0/16
gluetun    | |               ├── 169.254.0.0/16
gluetun    | |               ├── ::1/128
gluetun    | |               ├── fc00::/7
gluetun    | |               ├── fe80::/10
gluetun    | |               ├── ::ffff:127.0.0.1/104
gluetun    | |               ├── ::ffff:10.0.0.0/104
gluetun    | |               ├── ::ffff:169.254.0.0/112
gluetun    | |               ├── ::ffff:172.16.0.0/108
gluetun    | |               └── ::ffff:192.168.0.0/112
gluetun    | ├── Firewall settings:
gluetun    | |   ├── Enabled: yes
gluetun    | |   ├── VPN input ports:
gluetun    | |   |   └── 58930
gluetun    | |   └── Outbound subnets:
gluetun    | |       └── 10.5.0.0/16
gluetun    | ├── Log settings:
gluetun    | |   └── Log level: INFO
gluetun    | ├── Health settings:
gluetun    | |   ├── Server listening address: 127.0.0.1:9999
gluetun    | |   ├── Target address: cloudflare.com:443
gluetun    | |   ├── Duration to wait after success: 5s
gluetun    | |   ├── Read header timeout: 100ms
gluetun    | |   ├── Read timeout: 500ms
gluetun    | |   └── VPN wait durations:
gluetun    | |       ├── Initial duration: 6s
gluetun    | |       └── Additional duration: 5s
gluetun    | ├── Shadowsocks server settings:
gluetun    | |   └── Enabled: no
gluetun    | ├── HTTP proxy settings:
gluetun    | |   └── Enabled: no
gluetun    | ├── Control server settings:
gluetun    | |   ├── Listening address: :8000
gluetun    | |   └── Logging: yes
gluetun    | ├── OS Alpine settings:
gluetun    | |   ├── Process UID: 1000
gluetun    | |   ├── Process GID: 1000
gluetun    | |   └── Timezone: Europe/Berlin
gluetun    | ├── Public IP settings:
gluetun    | |   ├── Fetching: every 12h0m0s
gluetun    | |   └── IP file path: /tmp/gluetun/ip
gluetun    | └── Version settings:
gluetun    |     └── Enabled: yes
gluetun    | 2023-07-01T09:01:34+02:00 INFO [routing] default route found: interface eth0, gateway 172.18.0.1, assigned IP 172.18.0.26 and family v4
gluetun    | 2023-07-01T09:01:34+02:00 INFO [routing] default route found: interface eth0, gateway fdb3:cca5:e4ba:1000:d0cc:1:0:1, assigned IP 172.18.0.26 and family v6
gluetun    | 2023-07-01T09:01:34+02:00 INFO [routing] adding route for 0.0.0.0/0
gluetun    | 2023-07-01T09:01:34+02:00 INFO [routing] adding route for ::/0
gluetun    | 2023-07-01T09:01:34+02:00 INFO [firewall] setting allowed subnets...
gluetun    | 2023-07-01T09:01:34+02:00 INFO [routing] default route found: interface eth0, gateway 172.18.0.1, assigned IP 172.18.0.26 and family v4
gluetun    | 2023-07-01T09:01:34+02:00 INFO [routing] default route found: interface eth0, gateway fdb3:cca5:e4ba:1000:d0cc:1:0:1, assigned IP 172.18.0.26 and family v6
gluetun    | 2023-07-01T09:01:34+02:00 INFO [routing] adding route for 10.5.0.0/16
gluetun    | 2023-07-01T09:01:34+02:00 INFO [routing] adding route for 10.5.0.0/16
gluetun    | 2023-07-01T09:01:34+02:00 INFO [routing] routing cleanup...
gluetun    | 2023-07-01T09:01:34+02:00 INFO [routing] default route found: interface eth0, gateway 172.18.0.1, assigned IP 172.18.0.26 and family v4
gluetun    | 2023-07-01T09:01:34+02:00 INFO [routing] default route found: interface eth0, gateway fdb3:cca5:e4ba:1000:d0cc:1:0:1, assigned IP 172.18.0.26 and family v6
gluetun    | 2023-07-01T09:01:34+02:00 INFO [routing] deleting route for 0.0.0.0/0
gluetun    | 2023-07-01T09:01:34+02:00 INFO [routing] deleting route for ::/0
gluetun    | 2023-07-01T09:01:34+02:00 ERROR adding outbound subnet to routes: adding route for subnet 10.5.0.0/16: replacing route for subnet 10.5.0.0/16 at interface eth0: gateway, source, and destination ip are not the same IP family
gluetun    | 2023-07-01T09:01:34+02:00 INFO Shutdown successful`

With version v3.35 I have about the same log as @Lynnesbian.


My configuration:

```
version: "3"
services:
  gluetun:
    image: docker.io/qmcgaw/gluetun:v3.34.3
    container_name: gluetun
    # line above must be uncommented to allow external containers to connect. See https://github.com/qdm12/gluetun/wiki/Connect-a-container-to-gluetun#external-container-to-gluetun
    cap_add:
      - NET_ADMIN
    devices:
      - /dev/net/tun:/dev/net/tun
    ports:
      - 5800:5800     # Firefox
      - 8888:8888/tcp # HTTP proxy
      - 8388:8388/tcp # Shadowsocks
      - 8388:8388/udp # Shadowsocks
      - 51413:51413
      - 51413:51413/udp

    volumes:
      - ./data:/gluetun
    networks:
      - caddy_gluetunnet
      - caddy_proxy_net
    environment:
      - VPN_SERVICE_PROVIDER=mullvad
      - VPN_TYPE=wireguard
      - WIREGUARD_PRIVATE_KEY=[removed]
      - WIREGUARD_ADDRESSES=[removed]
      - SERVER_CITIES=[removed]
      # Timezone for accurate log times
      - TZ=Europe/Berlin
      - FIREWALL_OUTBOUND_SUBNETS=10.5.0.0/16
      - FIREWALL_VPN_INPUT_PORTS=58930
    restart: unless-stopped


networks:
  caddy_proxy_net:
    external: true
  caddy_gluetunnet:
    external: true
```

The 10.5.0.0/16 subnet is defined in my caddy docker-compose as follows:

```
networks:
[...]
  gluetunnet:
    driver: bridge
    ipam:
      config:
        - subnet: 10.5.0.0/16
          gateway: 10.5.0.1
```

**Comment by kevinkellner** [comment]
I have the same issue.

`VPN_PORT_FORWARDING=off` or `VPN_PORT_FORWARDING=on` does not make a difference. 

The error changes if I use v3.34.3:

```Creating gluetun ... done
Attaching to gluetun
gluetun    | ========================================
gluetun    | ========================================
gluetun    | =============== gluetun ================
gluetun    | ========================================
gluetun    | =========== Made with ❤️ by ============
gluetun    | ======= https://github.com/qdm12 =======
gluetun    | ========================================
gluetun    | ========================================
gluetun    | 
gluetun    | Running version v3.34.3 built on 2023-05-31T14:57:40.709Z (commit 42caa64)
gluetun    | 
gluetun    | 🔧 Need help? https://github.com/qdm12/gluetun/discussions/new
gluetun    | 🐛 Bug? https://github.com/qdm12/gluetun/issues/new
gluetun    | ✨ New feature? https://github.com/qdm12/gluetun/issues/new
gluetun    | ☕ Discussion? https://github.com/qdm12/gluetun/discussions/new
gluetun    | 💻 Email? quentin.mcgaw@gmail.com
gluetun    | 💰 Help me? https://www.paypal.me/qmcgaw https://github.com/sponsors/qdm12
gluetun    | 2023-07-01T09:01:33+02:00 INFO [routing] default route found: interface eth0, gateway 172.18.0.1, assigned IP 172.18.0.26 and family v4
gluetun    | 2023-07-01T09:01:33+02:00 INFO [routing] default route found: interface eth0, gateway fdb3:cca5:e4ba:1000:d0cc:1:0:1, assigned IP 172.18.0.26 and family v6
gluetun    | 2023-07-01T09:01:33+02:00 INFO [routing] local ethernet link found: eth1
gluetun    | 2023-07-01T09:01:33+02:00 INFO [routing] local ethernet link found: eth0
gluetun    | 2023-07-01T09:01:33+02:00 INFO [routing] local ipnet found: 10.5.0.0/16
gluetun    | 2023-07-01T09:01:33+02:00 INFO [routing] local ipnet found: 172.18.0.0/16
gluetun    | 2023-07-01T09:01:33+02:00 INFO [routing] local ipnet found: fdb3:cca5:e4ba:1000:d0cc:1::/112
gluetun    | 2023-07-01T09:01:33+02:00 INFO [routing] local ipnet found: fe80::/64
gluetun    | 2023-07-01T09:01:33+02:00 INFO [firewall] enabling...
gluetun    | 2023-07-01T09:01:33+02:00 INFO [firewall] enabled successfully
gluetun    | 2023-07-01T09:01:34+02:00 INFO [storage] merging by most recent 13056 hardcoded servers and 17657 servers read from /gluetun/servers.json
gluetun    | 2023-07-01T09:01:34+02:00 INFO [storage] Using mullvad servers from file which are 96 days more recent
gluetun    | 2023-07-01T09:01:34+02:00 INFO [storage] Using nordvpn servers from file which are 393 days more recent
gluetun    | 2023-07-01T09:01:34+02:00 INFO Alpine version: 3.17.3
gluetun    | 2023-07-01T09:01:34+02:00 INFO OpenVPN 2.4 version: 2.4.12
gluetun    | 2023-07-01T09:01:34+02:00 INFO OpenVPN 2.5 version: 2.5.8
gluetun    | 2023-07-01T09:01:34+02:00 INFO Unbound version: 1.17.1
gluetun    | 2023-07-01T09:01:34+02:00 INFO IPtables version: v1.8.8
gluetun    | 2023-07-01T09:01:34+02:00 INFO Settings summary:
gluetun    | ├── VPN settings:
gluetun    | |   ├── VPN provider settings:
gluetun    | |   |   ├── Name: mullvad
gluetun    | |   |   └── Server selection settings:
gluetun    | |   |       ├── VPN type: wireguard
gluetun    | |   |       ├── Cities: prague
gluetun    | |   |       └── Wireguard selection settings:
gluetun    | |   └── Wireguard settings:
gluetun    | |       ├── Private key: [removed]
gluetun    | |       ├── Interface addresses:
gluetun    | |       |   └── [removed]
gluetun    | |       └── Network interface: tun0
gluetun    | ├── DNS settings:
gluetun    | |   ├── DNS server address to use: 127.0.0.1
gluetun    | |   ├── Keep existing nameserver(s): no
gluetun    | |   └── DNS over TLS settings:
gluetun    | |       ├── Enabled: yes
gluetun    | |       ├── Update period: every 24h0m0s
gluetun    | |       ├── Unbound settings:
gluetun    | |       |   ├── Authoritative servers:
gluetun    | |       |   |   └── cloudflare
gluetun    | |       |   ├── Caching: yes
gluetun    | |       |   ├── IPv6: no
gluetun    | |       |   ├── Verbosity level: 1
gluetun    | |       |   ├── Verbosity details level: 0
gluetun    | |       |   ├── Validation log level: 0
gluetun    | |       |   ├── System user: root
gluetun    | |       |   └── Allowed networks:
gluetun    | |       |       ├── 0.0.0.0/0
gluetun    | |       |       └── ::/0
gluetun    | |       └── DNS filtering settings:
gluetun    | |           ├── Block malicious: yes
gluetun    | |           ├── Block ads: no
gluetun    | |           ├── Block surveillance: no
gluetun    | |           └── Blocked IP networks:
gluetun    | |               ├── 127.0.0.1/8
gluetun    | |               ├── 10.0.0.0/8
gluetun    | |               ├── 172.16.0.0/12
gluetun    | |               ├── 192.168.0.0/16
gluetun    | |               ├── 169.254.0.0/16
gluetun    | |               ├── ::1/128
gluetun    | |               ├── fc00::/7
gluetun    | |               ├── fe80::/10
gluetun    | |               ├── ::ffff:127.0.0.1/104
gluetun    | |               ├── ::ffff:10.0.0.0/104
gluetun    | |               ├── ::ffff:169.254.0.0/112
gluetun    | |               ├── ::ffff:172.16.0.0/108
gluetun    | |               └── ::ffff:192.168.0.0/112
gluetun    | ├── Firewall settings:
gluetun    | |   ├── Enabled: yes
gluetun    | |   ├── VPN input ports:
gluetun    | |   |   └── 58930
gluetun    | |   └── Outbound subnets:
gluetun    | |       └── 10.5.0.0/16
gluetun    | ├── Log settings:
gluetun    | |   └── Log level: INFO
gluetun    | ├── Health settings:
gluetun    | |   ├── Server listening address: 127.0.0.1:9999
gluetun    | |   ├── Target address: cloudflare.com:443
gluetun    | |   ├── Duration to wait after success: 5s
gluetun    | |   ├── Read header timeout: 100ms
gluetun    | |   ├── Read timeout: 500ms
gluetun    | |   └── VPN wait durations:
gluetun    | |       ├── Initial duration: 6s
gluetun    | |       └── Additional duration: 5s
gluetun    | ├── Shadowsocks server settings:
gluetun    | |   └── Enabled: no
gluetun    | ├── HTTP proxy settings:
gluetun    | |   └── Enabled: no
gluetun    | ├── Control server settings:
gluetun    | |   ├── Listening address: :8000
gluetun    | |   └── Logging: yes
gluetun    | ├── OS Alpine settings:
gluetun    | |   ├── Process UID: 1000
gluetun    | |   ├── Process GID: 1000
gluetun    | |   └── Timezone: Europe/Berlin
gluetun    | ├── Public IP settings:
gluetun    | |   ├── Fetching: every 12h0m0s
gluetun    | |   └── IP file path: /tmp/gluetun/ip
gluetun    | └── Version settings:
gluetun    |     └── Enabled: yes
gluetun    | 2023-07-01T09:01:34+02:00 INFO [routing] default route found: interface eth0, gateway 172.18.0.1, assigned IP 172.18.0.26 and family v4
gluetun    | 2023-07-01T09:01:34+02:00 INFO [routing] default route found: interface eth0, gateway fdb3:cca5:e4ba:1000:d0cc:1:0:1, assigned IP 172.18.0.26 and family v6
gluetun    | 2023-07-01T09:01:34+02:00 INFO [routing] adding route for 0.0.0.0/0
gluetun    | 2023-07-01T09:01:34+02:00 INFO [routing] adding route for ::/0
gluetun    | 2023-07-01T09:01:34+02:00 INFO [firewall] setting allowed subnets...
gluetun    | 2023-07-01T09:01:34+02:00 INFO [routing] default route found: interface eth0, gateway 172.18.0.1, assigned IP 172.18.0.26 and family v4
gluetun    | 2023-07-01T09:01:34+02:00 INFO [routing] default route found: interface eth0, gateway fdb3:cca5:e4ba:1000:d0cc:1:0:1, assigned IP 172.18.0.26 and family v6
gluetun    | 2023-07-01T09:01:34+02:00 INFO [routing] adding route for 10.5.0.0/16
gluetun    | 2023-07-01T09:01:34+02:00 INFO [routing] adding route for 10.5.0.0/16
gluetun    | 2023-07-01T09:01:34+02:00 INFO [routing] routing cleanup...
gluetun    | 2023-07-01T09:01:34+02:00 INFO [routing] default route found: interface eth0, gateway 172.18.0.1, assigned IP 172.18.0.26 and family v4
gluetun    | 2023-07-01T09:01:34+02:00 INFO [routing] default route found: interface eth0, gateway fdb3:cca5:e4ba:1000:d0cc:1:0:1, assigned IP 172.18.0.26 and family v6
gluetun    | 2023-07-01T09:01:34+02:00 INFO [routing] deleting route for 0.0.0.0/0
gluetun    | 2023-07-01T09:01:34+02:00 INFO [routing] deleting route for ::/0
gluetun    | 2023-07-01T09:01:34+02:00 ERROR adding outbound subnet to routes: adding route for subnet 10.5.0.0/16: replacing route for subnet 10.5.0.0/16 at interface eth0: gateway, source, and destination ip are not the same IP family
gluetun    | 2023-07-01T09:01:34+02:00 INFO Shutdown successful```

With version v3.35 I have about the same log as @Lynnesbian.


My configuration:

```
version: "3"
services:
  gluetun:
    image: docker.io/qmcgaw/gluetun:v3.34.3
    container_name: gluetun
    # line above must be uncommented to allow external containers to connect. See https://github.com/qdm12/gluetun/wiki/Connect-a-container-to-gluetun#external-container-to-gluetun
    cap_add:
      - NET_ADMIN
    devices:
      - /dev/net/tun:/dev/net/tun
    ports:
      - 5800:5800     # Firefox
      - 8888:8888/tcp # HTTP proxy
      - 8388:8388/tcp # Shadowsocks
      - 8388:8388/udp # Shadowsocks
      - 51413:51413
      - 51413:51413/udp

    volumes:
      - ./data:/gluetun
    networks:
      - caddy_gluetunnet
      - caddy_proxy_net
    environment:
      - VPN_SERVICE_PROVIDER=mullvad
      - VPN_TYPE=wireguard
      - WIREGUARD_PRIVATE_KEY=[removed]
      - WIREGUARD_ADDRESSES=[removed]
      - SERVER_CITIES=[removed]
      # Timezone for accurate log times
      - TZ=Europe/Berlin
      - FIREWALL_OUTBOUND_SUBNETS=10.5.0.0/16
      - FIREWALL_VPN_INPUT_PORTS=58930
    restart: unless-stopped


networks:
  caddy_proxy_net:
    external: true
  caddy_gluetunnet:
    external: true
```

The 10.5.0.0/16 subnet is defined in my caddy docker-compose as follows:

```
networks:
[...]
  gluetunnet:
    driver: bridge
    ipam:
      config:
        - subnet: 10.5.0.0/16
          gateway: 10.5.0.1
```

**Comment by NickSica** [comment]
I am having the same issues with the same error messages for each version.

**Comment by burkasaurusrex** [comment]
I have this issue as well, only after enabling ipv6

**Comment by qdm12** [comment]
Can someone run gluetun with `LOG_LEVEL=debug` that would help me 😉 
It's probably mixing an ipv6 gateway with the ipv4 destination `10.5.0.0/16`, but I would like to make sure that's the problem.

**Comment by qdm12** [comment]
39ae57f49d2b8175a8f12535bba0a857ed914340 should fix it on the latest image, I'll keep this issue opened, let me know if it resolves it 😉 

It now only adds routes/rules when the subnet ip family matches the found default route ip family.

**Comment by DareFox** [comment]
> [39ae57f](https://github.com/qdm12/gluetun/commit/39ae57f49d2b8175a8f12535bba0a857ed914340) should fix it on the latest image, I'll keep this issue opened, let me know if it resolves it 😉
> 
> It now only adds routes/rules when the subnet ip family matches the found default route ip family.

On latest version this error still exists

```
========================================
========================================
=============== gluetun ================
========================================
=========== Made with ❤️ by ============
======= https://github.com/qdm12 =======
========================================
========================================
Running version latest built on 2023-08-11T11:08:54.752Z (commit e556871)
🔧 Need help? https://github.com/qdm12/gluetun/discussions/new
🐛 Bug? https://github.com/qdm12/gluetun/issues/new
✨ New feature? https://github.com/qdm12/gluetun/issues/new
☕ Discussion? https://github.com/qdm12/gluetun/discussions/new
💻 Email? quentin.mcgaw@gmail.com
💰 Help me? https://www.paypal.me/qmcgaw https://github.com/sponsors/qdm12
2023-08-21T21:56:36Z INFO [routing] default route found: interface eth0, gateway 172.18.0.1, assigned IP 172.18.0.5 and family v4
2023-08-21T21:56:36Z INFO [routing] default route found: interface eth0, gateway fd00::2:0:0:1, assigned IP fd00::2:0:0:5 and family v6
2023-08-21T21:56:36Z INFO [routing] local ethernet link found: eth0
2023-08-21T21:56:36Z INFO [routing] local ipnet found: 172.18.0.0/16
2023-08-21T21:56:36Z INFO [routing] local ipnet found: fd00::2:0:0:0/80
2023-08-21T21:56:36Z INFO [routing] local ipnet found: fe80::/64
2023-08-21T21:56:36Z INFO [firewall] enabling...
2023-08-21T21:56:36Z DEBUG [firewall] iptables-nft --policy INPUT DROP
2023-08-21T21:56:36Z DEBUG [firewall] iptables-nft --policy OUTPUT DROP
2023-08-21T21:56:36Z DEBUG [firewall] iptables-nft --policy FORWARD DROP
2023-08-21T21:56:36Z DEBUG [firewall] ip6tables-nft --policy INPUT DROP
2023-08-21T21:56:36Z DEBUG [firewall] ip6tables-nft --policy OUTPUT DROP
2023-08-21T21:56:36Z DEBUG [firewall] ip6tables-nft --policy FORWARD DROP
2023-08-21T21:56:36Z DEBUG [firewall] iptables-nft --append INPUT -i lo -j ACCEPT
2023-08-21T21:56:36Z DEBUG [firewall] ip6tables-nft --append INPUT -i lo -j ACCEPT
2023-08-21T21:56:36Z DEBUG [firewall] iptables-nft --append OUTPUT -o lo -j ACCEPT
2023-08-21T21:56:36Z DEBUG [firewall] ip6tables-nft --append OUTPUT -o lo -j ACCEPT
2023-08-21T21:56:36Z DEBUG [firewall] iptables-nft --append OUTPUT -m conntrack --ctstate ESTABLISHED,RELATED -j ACCEPT
2023-08-21T21:56:36Z DEBUG [firewall] ip6tables-nft --append OUTPUT -m conntrack --ctstate ESTABLISHED,RELATED -j ACCEPT
2023-08-21T21:56:36Z DEBUG [firewall] iptables-nft --append INPUT -m conntrack --ctstate ESTABLISHED,RELATED -j ACCEPT
2023-08-21T21:56:36Z DEBUG [firewall] ip6tables-nft --append INPUT -m conntrack --ctstate ESTABLISHED,RELATED -j ACCEPT
2023-08-21T21:56:36Z DEBUG [firewall] iptables-nft --append OUTPUT -o eth0 -s 172.18.0.5 -d 172.18.0.0/16 -j ACCEPT
2023-08-21T21:56:36Z DEBUG [firewall] ip6tables-nft --append OUTPUT -o eth0 -d ff02::1:ff/104 -j ACCEPT
2023-08-21T21:56:36Z DEBUG [firewall] ip6tables-nft --append OUTPUT -o eth0 -s fd00::2:0:0:5 -d fd00::2:0:0:0/80 -j ACCEPT
2023-08-21T21:56:36Z DEBUG [firewall] ip6tables-nft --append OUTPUT -o eth0 -d ff02::1:ff/104 -j ACCEPT
2023-08-21T21:56:36Z DEBUG [firewall] ip6tables-nft --append OUTPUT -o eth0 -s fd00::2:0:0:5 -d fe80::/64 -j ACCEPT
2023-08-21T21:56:36Z DEBUG [firewall] ip6tables-nft --append OUTPUT -o eth0 -d ff02::1:ff/104 -j ACCEPT
2023-08-21T21:56:36Z DEBUG [firewall] iptables-nft --append INPUT -i eth0 -d 172.18.0.0/16 -j ACCEPT
2023-08-21T21:56:36Z DEBUG [firewall] ip6tables-nft --append INPUT -i eth0 -d fd00::2:0:0:0/80 -j ACCEPT
2023-08-21T21:56:36Z DEBUG [firewall] ip6tables-nft --append INPUT -i eth0 -d fe80::/64 -j ACCEPT
2023-08-21T21:56:36Z INFO [firewall] enabled successfully
2023-08-21T21:56:37Z INFO [storage] merging by most recent 17692 hardcoded servers and 17692 servers read from /gluetun/servers.json
2023-08-21T21:56:37Z INFO Alpine version: 3.18.3
2023-08-21T21:56:37Z INFO OpenVPN 2.5 version: 2.5.8
2023-08-21T21:56:37Z INFO OpenVPN 2.6 version: 2.6.5
2023-08-21T21:56:37Z INFO Unbound version: 1.17.1
2023-08-21T21:56:37Z INFO IPtables version: v1.8.9
2023-08-21T21:56:37Z INFO Settings summary:
├── VPN settings:
|   ├── VPN provider settings:
|   |   ├── Name: custom
|   |   └── Server selection settings:
|   |       ├── VPN type: wireguard
|   |       ├── Target IP address: [nuh uh]
|   |       └── Wireguard selection settings:
|   |           ├── Endpoint IP address: [nuh uh]
|   |           ├── Endpoint port: 61253
|   |           └── Server public key: +5085NqLnEjXSfVX9DGJMUo/mOv6+Lnz/fxvvfl3OgY=
|   └── Wireguard settings:
|       ├── Private key: [nuh uh]
|       ├── Pre-shared key: [nuh uh]
|       ├── Interface addresses:
|       |   ├── 10.66.66.3/32
|       |   └── fd42:42:42::3/128
|       ├── Allowed IPs:
|       |   ├── 0.0.0.0/0
|       |   └── ::/0
|       └── Network interface: tun0
|           └── MTU: 1400
├── DNS settings:
|   ├── Keep existing nameserver(s): no
|   ├── DNS server address to use: 127.0.0.1
|   └── DNS over TLS settings:
|       ├── Enabled: yes
|       ├── Update period: every 24h0m0s
|       ├── Unbound settings:
|       |   ├── Authoritative servers:
|       |   |   └── cloudflare
|       |   ├── Caching: yes
|       |   ├── IPv6: no
|       |   ├── Verbosity level: 1
|       |   ├── Verbosity details level: 0
|       |   ├── Validation log level: 0
|       |   ├── System user: root
|       |   └── Allowed networks:
|       |       ├── 0.0.0.0/0
|       |       └── ::/0
|       └── DNS filtering settings:
|           ├── Block malicious: yes
|           ├── Block ads: no
|           ├── Block surveillance: no
|           └── Blocked IP networks:
|               ├── 127.0.0.1/8
|               ├── 10.0.0.0/8
|               ├── 172.16.0.0/12
|               ├── 192.168.0.0/16
|               ├── 169.254.0.0/16
|               ├── ::1/128
|               ├── fc00::/7
|               ├── fe80::/10
|               ├── ::ffff:127.0.0.1/104
|               ├── ::ffff:10.0.0.0/104
|               ├── ::ffff:169.254.0.0/112
|               ├── ::ffff:172.16.0.0/108
|               └── ::ffff:192.168.0.0/112
├── Firewall settings:
|   ├── Enabled: yes
|   ├── Debug mode: on
|   ├── VPN input ports:
|   |   ├── 10333
|   |   └── 4321
|   └── Outbound subnets:
|       └── 192.168.0.0/16
├── Log settings:
|   └── Log level: INFO
├── Health settings:
|   ├── Server listening address: 127.0.0.1:9999
|   ├── Target address: cloudflare.com:443
|   ├── Duration to wait after success: 5s
|   ├── Read header timeout: 100ms
|   ├── Read timeout: 500ms
|   └── VPN wait durations:
|       ├── Initial duration: 6s
|       └── Additional duration: 5s
├── Shadowsocks server settings:
|   └── Enabled: no
├── HTTP proxy settings:
|   └── Enabled: no
├── Control server settings:
|   ├── Listening address: :8000
|   └── Logging: yes
├── OS Alpine settings:
|   ├── Process UID: 1000
|   └── Process GID: 1000
├── Public IP settings:
|   ├── Fetching: every 12h0m0s
|   └── IP file path: /tmp/gluetun/ip
└── Version settings:
    └── Enabled: yes
2023-08-21T21:56:37Z INFO [routing] default route found: interface eth0, gateway 172.18.0.1, assigned IP 172.18.0.5 and family v4
2023-08-21T21:56:37Z INFO [routing] default route found: interface eth0, gateway fd00::2:0:0:1, assigned IP fd00::2:0:0:5 and family v6
2023-08-21T21:56:37Z DEBUG [routing] ip rule add from 172.18.0.5/32 lookup 200 pref 100
2023-08-21T21:56:37Z DEBUG [routing] ip rule add from fd00::2:0:0:5/128 lookup 200 pref 100
2023-08-21T21:56:37Z INFO [routing] adding route for 0.0.0.0/0
2023-08-21T21:56:37Z DEBUG [routing] ip route replace 0.0.0.0/0 via 172.18.0.1 dev eth0 table 200
2023-08-21T21:56:37Z INFO [routing] adding route for ::/0
2023-08-21T21:56:37Z DEBUG [routing] ip route replace ::/0 via fd00::2:0:0:1 dev eth0 table 200
2023-08-21T21:56:37Z INFO [firewall] setting allowed subnets...
2023-08-21T21:56:37Z DEBUG [firewall] iptables-nft --append OUTPUT -o eth0 -s 172.18.0.5 -d 192.168.0.0/16 -j ACCEPT
2023-08-21T21:56:37Z DEBUG [firewall] ip6tables-nft --append OUTPUT -o eth0 -s fd00::2:0:0:5 -d 192.168.0.0/16 -j ACCEPT
2023-08-21T21:56:37Z INFO [routing] routing cleanup...
2023-08-21T21:56:37Z INFO [routing] default route found: interface eth0, gateway 172.18.0.1, assigned IP 172.18.0.5 and family v4
2023-08-21T21:56:37Z INFO [routing] default route found: interface eth0, gateway fd00::2:0:0:1, assigned IP fd00::2:0:0:5 and family v6
2023-08-21T21:56:37Z INFO [routing] deleting route for 0.0.0.0/0
2023-08-21T21:56:37Z DEBUG [routing] ip route delete 0.0.0.0/0 via 172.18.0.1 dev eth0 table 200
2023-08-21T21:56:37Z INFO [routing] deleting route for ::/0
2023-08-21T21:56:37Z DEBUG [routing] ip route delete ::/0 via fd00::2:0:0:1 dev eth0 table 200
2023-08-21T21:56:37Z DEBUG [routing] ip rule del from 172.18.0.5/32 lookup 200 pref 100
2023-08-21T21:56:37Z DEBUG [routing] ip rule del from fd00::2:0:0:5/128 lookup 200 pref 100
2023-08-21T21:56:37Z ERROR setting allowed outbound subnets: command failed: "ip6tables-nft --append OUTPUT -o eth0 -s fd00::2:0:0:5 -d 192.168.0.0/16 -j ACCEPT": ip6tables v1.8.9 (nf_tables): host/network `192.168.0.0' not found
Try `ip6tables -h' or 'ip6tables --help' for more information.: exit status 2
2023-08-21T21:56:37Z INFO Shutdown successful
```
### Docker Compose
```yaml
version: "3"
services:
  wireguard:
    privileged: true
    image: qmcgaw/gluetun:latest
    container_name: wireguard
    restart: unless-stopped
    ports:
      - "4321:4321" # port-checker
    networks: 
      - ipv6network
    volumes:
      - /docker/wireguard/config:/gluetun/wireguard/
    cap_add:
      - NET_ADMIN
    environment:
      - VPN_SERVICE_PROVIDER=custom
      - VPN_TYPE=wireguard
      - VPN_ENDPOINT_IP=[nuh uh]
      - VPN_ENDPOINT_PORT=63275
      - FIREWALL_VPN_INPUT_PORTS=10333,4321
      - FIREWALL_DEBUG=on
      - FIREWALL_OUTBOUND_SUBNETS=192.168.0.0/16
      - WIREGUARD_PUBLIC_KEY=+5085NqLnEjXSfVX9DGJMUo/mOv6+Lnz/fxvvfl3OgY=
      - WIREGUARD_PRIVATE_KEY=[nuh uh]
      - WIREGUARD_ADDRESSES=10.66.66.3/32,fd42:42:42::3/128
      - WIREGUARD_PRESHARED_KEY=[nuh uh]

networks:
  ipv6network:
    enable_ipv6: true
    driver: bridge
    ipam:
      driver: default
      config:
        - subnet: fd00:0:0:0:2::/80
```

### wg0.conf

```ini
[Interface]
PrivateKey = [nuh uh]
Address = 10.66.66.3/32,fd42:42:42::3/128
DNS = 1.1.1.1,1.0.0.1

[Peer]
PublicKey = 4cR4CR9AdpCAswZ42lFMTWVLCwzfhtG3V0vYY+lsUh8=
PresharedKey = [nuh uh]
Endpoint = [nuh uh]:63275
AllowedIPs = 0.0.0.0/0,::/0
```

**Comment by samip5** [comment]
This is still occuring with container `ghcr.io/qdm12/gluetun:latest@sha256:8d744532dc4a2394f9b6544637b1c1312be8e38903777df6081fd1511d5d1223`

**Comment by kevinkellner** [comment]
Can confirm that setting FIREWALL_OUTBOUND_SUBNETS works now. Thank you very much.

**Comment by qdm12** [comment]
Awesome, thanks @kevinkellner for confirming. If anything related happens again, please create another issue (I'm not monitoring closed issues). Thanks!!
